### ISSUE TITLE
Model base class is a little too restrictive

### ISSUE THREAD

**Issue by mortonjt** [gibsramen/BIRDMAn#34]
I'm trying to design a new stan model that doesn't take as input a formula, but instead passes in 2 metadata columns.

Namely, it has the following signature
```python
class NegativeBinomialCaseControl(birdman.Model):
    def __init__(self,
                 table: biom.table.Table,
                 status_column : str,
                 matching_column : str,
                 metadata: pd.DataFrame,
                 num_iter: int = 500,
                 num_warmup: int = None,
                 adapt_delta: float = 0.9,
                 max_treedepth: float = 20,
                 chains: int = 4,
                 seed: float = 42,
                 mu_scale: float = 10,
                 sigma_scale: float = 1,
                 disp_scale: float = 1,
                 control_scale: float = 10):
        super(NegativeBinomialCaseControl, self).__init__(
            table, status_column, matching_column,
            metadata, num_iter, num_warmup, adapt_delta,
            max_treedepth, chains, seed,
            mu_scale, sigma_scale, disp_scale, control_scale,
        )

```
If I try to build this model, I will get the following error
```
Traceback (most recent call last):
  File "test_stan.py", line 91, in test_cc
    seed=42)
  File "/home/juermieboop/Documents/research/q2/q2-differential/q2_differential/_stan.py", line 197, in __init__
    super(NegativeBinomialCaseControl, self).__init__()
TypeError: __init__() missing 4 required positional arguments: 'table', 'formula', 'metadata', and 'model_path'
```
The existing `Model` is no doubt useful, but perhaps we can rename this to `RegressionModel`, since those are the types of models this currently encapsulates. I'm also wondering if we could create an additional abstract class to handle more broad use cases such as case-control matching.  I'll quickly sketch out a possibility of what this class will look like shortly.

**Comment by gibsramen** [comment]
This is a good point, thanks for bringing it up. Definitely agree that we could abstract out the base class more.

**Comment by mortonjt** [comment]
I think this has been addressed by #38, so closing.
